File: kotlin-pr/50/kotlin/code-quality-logging/SecureRequestDebugger.kt
data class SecureRequest(
    val url: String,
    val method: String,
    val headers: Map<String, String>,
    val body: String,
    val session: Map<String, Any>?
)

class SecureRequestDebugger {
    fun debugRequest(request: SecureRequest) {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        println("Request URL: ${request.url}")
        println("Request method: ${request.method}")
        println("Content-Type: ${request.headers["Content-Type"] ?: "not set"}")
        println("Request size: ${request.body.length} bytes")
        println("Session active: ${if (request.session != null) "yes" else "no"}")
        // {/fact}
    }
}

File: scala-pr/54/scala/code-quality-logging/SecureRequestDebugger.scala
case class SecureRequest(url: String, method: String, headers: Map[String, String], body: String, session: Option[Map[String, Any]])

class SecureRequestDebugger {
  def debugRequest(request: SecureRequest): Unit = {
    // {fact rule=code-quality-logging@v1.0 defects=0}
    println(s"Request URL: ${request.url}")
    println(s"Request method: ${request.method}")
    println(s"Content-Type: ${request.headers.getOrElse("Content-Type", "not set")}")
    println(s"Request size: ${request.body.length} bytes")
    println(s"Session active: ${if (request.session.isDefined) "yes" else "no"}")
    // {/fact}
  }
}